ext/mysql2/extconf.rb

```diff
@@ -22,7 +22,13 @@ def add_ssl_defines(header)
 end
 
 # Homebrew openssl
-$LDFLAGS << ' -L/usr/local/opt/openssl/lib' if RUBY_PLATFORM =~ /darwin/
+brew_installed = system("command -v brew") 
+if brew_installed
+  openssl_location = `brew --prefix openssl`.strip
+  if RUBY_PLATFORM =~ /darwin/ && openssl_location
+    $LDFLAGS << " -L#{openssl_location}/lib" 
+  end
+end
```

Comment:
Thank you for this PR!

Would it make sense to invert the checks so that only if RUBY_PLATFORM is darwin then brew is checked for?

Reply:
It really does, simplified the conditional a bit too.